gcloud/datastore/connection.py

```diff
@@ -312,6 +312,7 @@ def commit(self, dataset_id, mutation_pb, transaction_id):
                    that was completed in the commit.
         """
         request = _datastore_pb2.CommitRequest()
+        request.CopyFrom(commit_request)
```

Comment:
I don't think we ever contemplated reusing `Batch`/`Transaction` instances after committing them:  makes me queasy to think about it.